Improved docs for Loggers

awaelchli opened the pull request:
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

<!-- For CHANGELOG separate each item in unreleased section by blank line to reduce collisions -->

## What does this PR do?
A continuation of a series of doc improvements. Follows #1389, #1370, #1483 

- typos
- added links for classes, methods, modules etc. for easier navigation within the page.
- converted many code snippets to doctests and fixed failing ones.
- added instructions how to install the loggers #1487.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

pep8speaks commented:
Hello @awaelchli! Thanks for opening this PR. 



* In the file [`pytorch_lightning/loggers/base.py`](https://github.com/PyTorchLightning/pytorch-lightning/blob/1aae1ac2f3b75d612eab2051f17c2144fc8e930f/pytorch_lightning/loggers/base.py):

> [Line 272:111](https://github.com/PyTorchLightning/pytorch-lightning/blob/1aae1ac2f3b75d612eab2051f17c2144fc8e930f/pytorch_lightning/loggers/base.py#L272): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (112 > 110 characters)





* In the file [`pytorch_lightning/loggers/neptune.py`](https://github.com/PyTorchLightning/pytorch-lightning/blob/1aae1ac2f3b75d612eab2051f17c2144fc8e930f/pytorch_lightning/loggers/neptune.py):

> [Line 307:111](https://github.com/PyTorchLightning/pytorch-lightning/blob/1aae1ac2f3b75d612eab2051f17c2144fc8e930f/pytorch_lightning/loggers/neptune.py#L307): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (114 > 110 characters)

* In the file [`pytorch_lightning/loggers/tensorboard.py`](https://github.com/PyTorchLightning/pytorch-lightning/blob/1aae1ac2f3b75d612eab2051f17c2144fc8e930f/pytorch_lightning/loggers/tensorboard.py):

> [Line 74:111](https://github.com/PyTorchLightning/pytorch-lightning/blob/1aae1ac2f3b75d612eab2051f17c2144fc8e930f/pytorch_lightning/loggers/tensorboard.py#L74): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (114 > 110 characters)







Do see the [Hitchhiker's guide to code style](https://goo.gl/hqbW4r)

Borda commented on docs/source/experiment_logging.rst:
```diff
@@ -1,201 +1,186 @@
 Experiment Logging
-===================
+==================
```
not sure but I think that doctest in RST files are not executed, so there is no need of converting them here
https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html

Borda commented on pytorch_lightning/loggers/mlflow.py:
```diff
@@ -39,17 +19,38 @@ def any_lightning_module_function_or_hook(...):
 class MLFlowLogger(LightningLoggerBase):
+    """
+    Log using `MLflow <https://mlflow.org>`_
+
```
```suggestion
    Example:
    
```

Borda commented on pytorch_lightning/loggers/comet.py:
```diff
@@ -33,57 +30,56 @@
+    Log using `Comet.ml <https://www.comet.ml>`_.
+    Requires either an API Key (online mode) or a local directory path (offline mode).
+
+    **ONLINE MODE**
```
```suggestion
    Example:
    **ONLINE MODE**
```

Borda replied:
as example section, the doctest shall be shifted by 4 spaces...

Borda commented on pytorch_lightning/loggers/trains.py:
```diff
@@ -44,24 +23,45 @@ def any_lightning_module_function_or_hook(...):
-    """Logs using TRAINS
+    """
+    Log using `allegro.ai TRAINS <https://github.com/allegroai/trains>`_.
+
```
```suggestion

    Example:
```

awaelchli commented:
@Borda 
Good point about doctest. I was wondering. 
I can run it locally like this for example: 
`python -m doctest docs/source/experiment_logging.rst`
Maybe we could add it to CI like this?
I will check the links you posted.

Borda commented:
sure add it to CI :]

awaelchli commented:
Hey @jakubczakon do you know how I could make the doctest pass in online mode without complaining about an invalid api key?

jakubczakon commented:
Hi @awaelchli,

I think the reason could be:

- the project is not specified correctly. `api_token="ANONYMOUS"` works for some projects in the "shared" organization. We need to add that project by hand. That said "shared/pytorch-lightning-integration" is working.
- you are using an older version of neptune-client that doesn't yet have "ANONYMOUS" token hardcoded -> updating the project should fix it

Also, the fact that `api_token="test"` works in offline mode is because in offline mode nothing gets sent to Neptune so no token is checked etc.

Could you please paste the particular piece of code that is causing the problem?

awaelchli commented:
@jakubczakon AWESOME. Your tipp with the version update did the trick! I didn't see it fail locally because I had the new version and the CI tests with minimal requirements. 
Thanks for your help!!

awaelchli commented:
@Borda The doctest extension is already in `conf.py`. It can be run with `make doctest`. If I do locally, I get many import errors. Let's maybe fix it in another PR?

Borda commented:
sounds fair to do it extra PR... I would maybe run in pytest then sphinx...

Borda commented:
@awaelchli rebase to master shall fix it...

Borda commented on pytorch_lightning/loggers/mlflow.py:
```diff
@@ -40,21 +19,46 @@ def any_lightning_module_function_or_hook(...):
+
+    Example:
+
+    >>> from pytorch_lightning import Trainer
```
this shall be shifted by 4 spaces

awaelchli replied:
is it ok now?

Borda replied:
it is for this one but the same is in two more places
as Example is a section, the doctest shall be shifted, right?

awaelchli replied:
ok, right. check again, is it better now?

awaelchli commented:
@Borda about RST files: could I do that in another PR? on master "make doctest" has many errors (including RST) that would need to be solved and I think it would be better to do in another PR. What do you think?
https://github.com/PyTorchLightning/pytorch-lightning/pull/1484#issuecomment-613546668

Borda commented:
so pls make it an issue linked to #714  so we do not forget... 
because now it seems as these examples shall be tested (because they are formatted as doctest) but in fact, they are not... lol

awaelchli commented:
I already added a note in #714 as a reminder 2 days ago. 
and the tests that I added here in rst I tested locally, with "make doctest docs/source/file.rst"
